### PR TITLE
Upgrades GODMODE from a flag to a trait.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
GODMODE has a lot of sources that toggle it. From admin-stuff to status effects, components, actions and mobs which are supposed to be invincible. It's better off as a trait than a flag, so we can manage these sources.

## Why It's Good For The Game
See above.

## Changelog

:cl:
admin: godmode is now a datum trait instead of a bitflag. This means the process for toggling it is a little different now.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 1]
```diff
@@ -1,13 +1,16 @@
 
 /mob/living/carbon/human/dummy
 	real_name = "Test Dummy"
-	status_flags = GODMODE|CANPUSH
```
Comment (SuperNovaa41) [inline code/modules/mob/living/carbon/human/dummy.dm:4]: did you mean to remove the `CANPUSH` flag ?

Comment (ZephyrTFA) [reply code/modules/mob/living/carbon/human/dummy.dm:4]: Why does a dummy need can push

Comment (SuperNovaa41) [reply code/modules/mob/living/carbon/human/dummy.dm:4]: i have no idea

Comment (Ghommie) [reply code/modules/mob/living/carbon/human/dummy.dm:4]: Keep in mind that it just inherits the status flags of the parent type in this case.

File: code/modules/mob/living/basic/space_fauna/bear/_bear.dm [hunk 2]
```diff
@@ -94,12 +94,11 @@
 /mob/living/basic/bear/snow/ancient
 	name = "ancient polar bear"
 	desc = "A grizzled old polar bear, its hide thick enough to make it impervious to almost all weapons."
-	status_flags = CANPUSH | GODMODE
 	gold_core_spawnable = NO_SPAWN
 
 /mob/living/basic/bear/snow/Initialize(mapload)
```
Comment (iprice) [inline code/modules/mob/living/basic/space_fauna/bear/_bear.dm:99]: Moved godmode from bear/snow/ancient to bear/snow

File: code/modules/mob/living/basic/pets/parrot/poly.dm [hunk 3]
```diff
@@ -191,7 +191,6 @@
 	name = "The Ghost of Poly"
 	desc = "Doomed to squawk the Earth."
 	color = "#FFFFFF77"
-	status_flags = GODMODE
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/pets/parrot/poly.dm:194]: Should be `status_flags = NONE`

File: code/modules/mob/camera/camera.dm [hunk 4]
```diff
@@ -4,7 +4,6 @@
 	density = FALSE
 	move_force = INFINITY
 	move_resist = INFINITY
-	status_flags = GODMODE  // You can't damage it.
```
Comment (MrMelbert) [inline code/modules/mob/camera/camera.dm:7]: Should be `status_flags = NONE`

File: code/modules/mob/living/basic/ruin_defender/living_floor.dm [hunk 5]
```diff
@@ -26,7 +26,6 @@
 	icon_living = "floor"
 	mob_size = MOB_SIZE_HUGE
 	mob_biotypes = MOB_SPECIAL
-	status_flags = GODMODE //nothing but crowbars may kill us
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/ruin_defender/living_floor.dm:29]: Should be `status_flags = NONE`

File: code/modules/mob/living/basic/space_fauna/bear/_bear.dm [hunk 6]
```diff
@@ -94,12 +94,11 @@
 /mob/living/basic/bear/snow/ancient
 	name = "ancient polar bear"
 	desc = "A grizzled old polar bear, its hide thick enough to make it impervious to almost all weapons."
-	status_flags = CANPUSH | GODMODE
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/space_fauna/bear/_bear.dm:97]: Should be `status_flags = CANPUSH`

File: code/modules/mob/living/basic/pets/parrot/poly.dm (hunk 3 above)
Comment (Time-Green) [reply code/modules/mob/living/basic/pets/parrot/poly.dm:194]: Can’t believe ghommie would do this

File: code/modules/mob/living/basic/space_fauna/bear/_bear.dm (hunk 2 above)
Comment (Ghommie) [reply code/modules/mob/living/basic/space_fauna/bear/_bear.dm:99]: OOOOH FUK

**Pull request merged**